datastream-kafka/src/main/java/com/linkedin/datastream/kafka/KafkaProducerWrapper.java

```diff
@@ -246,7 +253,7 @@ private synchronized void shutdownProducer() {
     }
   }
 
-  private DatastreamRuntimeException generateSendFailure(Exception exception) {
+  protected DatastreamRuntimeException generateSendFailure(Exception exception) {
```

Comment:
I don't see you overriding this in your Bounded implementation, why make this protected?